packages/shared/src/porter.ts

```diff
@@ -120,10 +137,35 @@ export type TacoDecryptResult = {
 };
 
 export class PorterClient {
-  readonly porterUrl: URL;
+  readonly porterUrls: URL[];
 
-  constructor(porterUri: string) {
-    this.porterUrl = new URL(porterUri);
+  constructor(porterUris: string | string[]) {
+    if (porterUris instanceof Array) {
+      this.porterUrls = porterUris.map((uri) => new URL(uri));
+    } else {
+      this.porterUrls = [new URL(porterUris)];
+    }
+  }
+
+  protected async tryAndCall<T, D>(config: AxiosRequestConfig<D>): Promise<AxiosResponse<T>> {
+    let resp!: AxiosResponse<T>;
+    let lastError = undefined;
+    for (const porterUrl of this.porterUrls) {
+      const localConfig = { ...config, baseURL: porterUrl.toString() }
+      try {
+        resp = await axios.request(localConfig);
+      } catch (e) {
+        lastError = e;
+        continue;
+      }
+      if (resp.status === HttpStatusCode.Ok) {
+        return resp;
+      }
+    }
+    if (lastError !== undefined) {
+      throw lastError;
+    }
+    throw new Error("Porter returns bad response");
```

Comment:
So this line only runs if Porter returned an HTTP code different than `200 OK`? But no exception was thrown.

I think it would be helpful to somehow propagate the response from Porter in this exception, maybe like so:
```suggestion
    throw new Error(`Porter returns bad response: ${resp.status} - ${resp.message}`);
```
This wouldn't work on this particular line, but maybe we can squeeze it into an `else` statement after line 163.

Just throwing ideas, WDYT?

Reply:
Actually it will work because resp at this line won't be empty and it will be last bad response
✔️